packages/expo-cli/src/commands/upload/IOSUploader.ts

```diff
@@ -110,16 +121,19 @@ export default class IOSUploader extends BaseUploader {
     };
   }
 
-  async _getAppleTeamId() {
+  async _getAppleTeamId(): Promise<string | undefined> {
     const credentialMetadata = await Credentials.getCredentialMetadataAsync(this.projectDir, 'ios');
-    const { teamId } = await Credentials.getCredentialsForPlatform(credentialMetadata);
+    const credential = await Credentials.getCredentialsForPlatform(credentialMetadata);
 
-    return teamId;
+    if (credential) {
+      return credential.teamId;
+    }
+    return;
   }
 
-  async _getAppleIdCredentials() {
+  async _getAppleIdCredentials(): Promise<IosPlatformOptions> {
     const appleCredsKeys = ['appleId', 'appleIdPassword'];
-    const result = pick(this.options, appleCredsKeys);
+    const result: any = pick(this.options, appleCredsKeys);
```

Comment:
It's tricky that the type definitions of lodash make the return type `Partial< IosPlatformOptions>` here, which is too broad. Using `any` here is not ideal, it masked the error that showed the return type of this function was wrong.

Maybe an explicit cast would be slightly better?
```suggestion
    const result: any = pick(this.options, appleCredsKeys) as Pick< IosPlatformOptions, 'appleId' | 'appleIdPassword'>;
```
or maybe drop the `lodash.pick`, since  it gives a lot of TypeScript troubles, but little benefit in a simple case like this:
```suggestion
    const result = { appleId: this.options.appleId, appleIdPassword: this.options. appleIdPassword };
```
I think I actually prefer the latter.